addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -359,13 +490,188 @@ odoo.define('website_slides.quiz', function (require) {
                 values.push(parseInt($(inputs[i]).val()));
             }
 
-            if (values.length === this.quiz.questions.length){
-                this._alertHide();
+            if (values.length === this.quiz.questionsCount){
                 this._submitQuiz(values);
             } else {
                 this._alertShow();
             }
         },
+
+        /**
+         * When clicking on 'Add a Question' or 'Add Quiz' it
+         * initialize a new QuestionFormWidget to input the new
+         * question.
+         * @private
+         */
+        _onCreateQuizClick: function () {
+            var $elem = this.$('.o_wslides_js_lesson_quiz_new_question');
+            this.$('.o_wslides_js_quiz_add').addClass('d-none');
+            new QuestionFormWidget(this, {
+                slideId: this.slide.id,
+                sequence: this.quiz.questionsCount + 1
+            }).appendTo($elem);
+        },
+
+        /**
+         * When clicking on the edit button of a question it
+         * initialize a new QuestionFormWidget with the existing
+         * question as inputs.
+         * @param ev
+         * @private
+         */
+        _onEditQuestionClick: function (ev) {
+            var $elem = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            var question = this._getQuestionDetails($elem);
+            new QuestionFormWidget(this, {
+                oldElement: $elem,
+                question: question,
+                slideId: this.slide.id,
+                sequence: question.sequence,
+                update: true
+            }).insertAfter($elem);
+            $elem.hide();
+        },
+
+        /**
+         * When clicking on the delete button of a question it
+         * toggles a modal to confirm the deletion
+         * @param ev
+         * @private
+         */
+        _onDeleteQuestionClick: function (ev) {
+            var question = $(ev.currentTarget).closest('.o_wslides_js_lesson_quiz_question');
+            new ConfirmationDialog(this, {
+                questionId: question.data('questionId'),
+                questionTitle: question.data('title')
+            }).open();
+        },
+
+        /**
+         * Displays the created Question at the correct place (after the last question or
+         * at the first place if there is no questions yet) It also displays the 'Add Question'
+         * button or open a new QuestionFormWidget if the user wants to immediately add another one.
+         *
+         * @param event
+         * @private
+         */
+        _displayCreatedQuestion: function (event) {
+            var $lastQuestion = this.$('.o_wslides_js_lesson_quiz_question:last');
+            if ($lastQuestion.length !== 0) {
+                $lastQuestion.after(event.data.view);
+            } else {
+                this.$el.prepend(event.data.view);
+            }
+            this.quiz.questionsCount++;
+            event.data.widget.destroy();
```

Comment:
This is nice, but maybe find a better name than just `widget` ?

Something like `questionFormWidget` ?